docs/dev/security-architecture.md

```diff
@@ -20,4 +20,58 @@ password, it will response authentication result with authentication Id.
  Users can defined their own JAAS Login module or write own custom authenticators as well.
   - [Changing the Default Security Settings](../user/authentication-settings-guide.md)
 
-If authentication is success, connection will be established or else will send authentication error.
\ No newline at end of file
+If authentication is success, connection will be established or else will send authentication error.
+
+### Writing a custom authenticator
+
+Custom authenticator should implement the [Authenticator](https://github.com/ballerina-platform/ballerina-message-broker/blob/master/modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/Authenticator.java) interface and handle the 
+authentication underlying logic. Message broker only interested in [AuthResult](https://github.com/ballerina-platform/ballerina-message-broker/blob/master/modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/AuthResult.java) 
```

Comment:
`Message broker only interested ` -> `Message broker is only interested` ?

Reply:
Fixed in 41f8861